src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ClickOnce.targets

```diff
@@ -0,0 +1,86 @@
+  </PropertyGroup>
+
+  <!-- 
+    For .NET Core ClickOnce publish in Single File mode, the ClickOnce manifest generation needs to run 
+    after ComputeFilesToPublish. This is so that the bundle EXE which is included in the ClickOnce manifest 
+    is generated before the manifest is created.
+  -->
+  <PropertyGroup Condition="'$(PublishSingleFile)' == 'true'">
+    <DeploymentComputeClickOnceManifestInfoDependsOn>
+      $(DeploymentComputeClickOnceManifestInfoDependsOn);
+      ComputeFilesToPublish
+    </DeploymentComputeClickOnceManifestInfoDependsOn>
+  </PropertyGroup>
+
+  <!-- Ensure runtimeconfig.json is built before clickonce manifest generation -->
+  <PropertyGroup Condition="'$(GenerateRuntimeConfigurationFiles)'=='true'">
+    <GenerateManifestsDependsOn>
+       GenerateBuildRuntimeConfigurationFiles;
+       $(GenerateManifestsDependsOn)
+    </GenerateManifestsDependsOn>
+  </PropertyGroup>
+
+  <!-- Ensure deps.json is available before clickonce manifest generation -->
+  <PropertyGroup Condition="'$(GenerateDependencyFile)' == 'true'">
+    <GenerateManifestsDependsOn>
+      _ComputeUseBuildDependencyFile;
+      GenerateBuildDependencyFile;
+      GeneratePublishDependencyFile;
+      GetNetCoreRuntimeJsonFilesForClickOnce;
+      $(GenerateManifestsDependsOn)
+    </GenerateManifestsDependsOn>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(GenerateRuntimeConfigurationFiles)'=='true'">
+    <ProjectRuntimeConfigFilesForClickOnce Include="$(ProjectRuntimeConfigFilePath)"/>
+  </ItemGroup>
+
+  <Target Name="GetNetCoreRuntimeJsonFilesForClickOnce" 
+          Condition="'$(GenerateDependencyFile)' == 'true'">
+
+    <!-- Get correct deps json files based on _UseBuildDependencyFile -->
+    <ItemGroup>
+      <ProjectDepsFilesForClickOnce Condition="'$(_UseBuildDependencyFile)' == 'true'" Include="$(ProjectDepsFilePath)"/>
+      <ProjectDepsFilesForClickOnce Condition="'$(_UseBuildDependencyFile)' != 'true'" Include="$(PublishDepsFilePath)"/>
+    </ItemGroup>
+
+    <!-- Add runtimeconfig and deps json file to item group that's included in files for clickonce publishing -->
+    <ItemGroup>
+       <NetCoreRuntimeJsonFilesForClickOnce Include="@(ProjectRuntimeConfigFilesForClickOnce);@(ProjectDepsFilesForClickOnce)"/>
+    </ItemGroup>
+  </Target>  
+
+  <!-- 
+    Add necessary clickonce targets that need to run during publish process.
+    These targets are defined in MS.Common.CurrentVersion.targets in the msbuild repo.
+  -->
+  <PropertyGroup>
+    <ClickOncePublishDependsOn>
+      $(ClickOncePublishDependsOn);
+      _CopyFilesToPublishFolder;
+      _DeploymentGenerateBootstrapper;
+      ResolveKeySource;
+      _DeploymentSignClickOnceDeployment;
+    </ClickOncePublishDependsOn>
+  </PropertyGroup>
+
+  <Target Name="ClickOncePublish" Condition="'$(PublishableProject)'=='true'" DependsOnTargets="$(ClickOncePublishDependsOn)" />
```

Comment:
is it done? What is the target to do the package generation?

Reply:
ClickOncePublish is a pseudo target that has dependent targets set by ClickOncePublishDependsOn. These dependent targets are defined in the MS.Common.CurrentVersion.targets file. The PublishProtocolProviderTargets property is an extension point defined in MS.Common.CurrentVersion.targets and we set this property's value to ClickOncePublish. This way we inject the dependent targets that we need to invoke during the .NET Core Publish process.